cmd/docker-mcp/internal/gateway/handlers.go

```diff
@@ -2,16 +2,41 @@ package gateway
 
 import (
 	"context"
+	"fmt"
+	"os"
+	"time"
 
 	"github.com/modelcontextprotocol/go-sdk/mcp"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/metric"
 
 	"github.com/docker/mcp-gateway/cmd/docker-mcp/internal/catalog"
+	"github.com/docker/mcp-gateway/cmd/docker-mcp/internal/telemetry"
 )
 
 func getClientConfig(readOnlyHint *bool, ss *mcp.ServerSession, server *mcp.Server) *clientConfig {
 	return &clientConfig{readOnly: readOnlyHint, serverSession: ss, server: server}
 }
 
+// inferServerType determines the type of MCP server based on its configuration
+func inferServerType(serverConfig *catalog.ServerConfig) string {
+	// Check for SSE endpoint first (including remote URLs)
+	if serverConfig.Spec.SSEEndpoint != "" || serverConfig.Spec.Remote.URL != "" {
+		return "sse"
+	}
+	// Check for Docker image
+	if serverConfig.Spec.Image != "" {
```

Comment:
docker sort of implies "stdio" today because there are no examples of containers that force us to connect to them over anything other than stdio. It's not impossible that there could be a docker server type which was non-stdio.  It just hasn't happened (and doesn't really make a lot of sense)

Reply:
What do you think of:

- streaming
- sse
- docker (pre-packaged image + stdio)
- docker-jit (JIT package + stdio)
- stdio (when/if we ever do vanilla stdio against raw commands; should be zero of these today)